modules/generic-extras/src/main/scala/io/circe/generic/extras/Configuration.scala

```diff
@@ -40,17 +42,21 @@ final case class Configuration(
 }
 
 final object Configuration {
+
   val default: Configuration = Configuration(Predef.identity, Predef.identity, false, None)
+  val basePattern: Pattern = Pattern.compile("([A-Z]+)([A-Z][a-z])")
```

Comment:
Could you make these private? They shouldn't be needed outside of the implementations.

Reply:
Here are fast implementations of enforcing transformations for snake-case, kebab-case, and camelCase: https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-macros/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/macros/JsonCodecMaker.scala#L52-L98

Please, fill free to adopt them for your requirements.